library/core/src/str/pattern.rs

```diff
@@ -1940,3 +1947,17 @@ unsafe fn small_slice_eq(x: &[u8], y: &[u8]) -> bool {
         vx == vy
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod verify {
+    #[cfg(all(target_arch = "x86_64", target_feature = "sse2"))] // only called on x86
+    #[kani::proof]
+    pub fn check_small_slice_eq() {
+        let _ = Box::new(0);
+        const ARR_SIZE: usize = 1000;
+        let x: [i32; ARR_SIZE] = kani::any();
+        let y: [i32; ARR_SIZE] = kani::any();
+        small_slice_eq(x, y);
```

Comment:
Don't you need an unsafe block here?

Reply:
Yes. I didn't find a good way to test it. Now I am running it by dropping the cfg constraint `target="sse2"`.